### PR TITLE
Implement externalScored attribute in outcome declaration

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
Attribute externalScored added to OutcomeDeclaration element

https://oat-sa.atlassian.net/browse/MS-59

### CODE REVIEW

File: src/qtism/data/state/ExternalScore.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        switch ($constant) {
+            case self::HUMAN:
+                return 'human';
+            case self::EXTERNAL_MACHINE:
+                return 'externalMachine';
+            default:
+                return false;
+        }
+    }
+}
```
Comment (epoxxid) [inline src/qtism/data/state/ExternalScore.php]: Missing empty line

File: src/qtism/data/storage/xml/marshalling/OutcomeDeclarationMarshaller.php [hunk 2]
```diff
@@ -115,6 +119,11 @@ protected function unmarshall(DOMElement $element)
             $object->setCardinality($baseComponent->getCardinality());
             $object->setDefaultValue($baseComponent->getDefaultValue());
 
+            // Set external scored attribute
+            if ($element->hasAttribute('externalScored')) {
```
Comment (epoxxid) [inline src/qtism/data/storage/xml/marshalling/OutcomeDeclarationMarshaller.php]: Here already `externalScored` :) How could you make it more consistent?

File: test/samples/ims/items/2_2/essay.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        </p>
+        <extendedTextInteraction responseIdentifier="RESPONSE">
+            <prompt>
+                <b>
+                    Write an abstract about the life of this historical figure.
+                </b>
+            </prompt>
+        </extendedTextInteraction>
+    </itemBody>
+</assessmentItem>
```
Comment (epoxxid) [inline test/samples/ims/items/2_2/essay.xml]: Empty line exptected

File: src/qtism/data/state/ExternalScore.php (hunk 1 above)
Comment (bartlomiejmarszal) [inline src/qtism/data/state/ExternalScore.php]:
```suggestion
}

```

File: src/qtism/data/storage/xml/marshalling/OutcomeDeclarationMarshaller.php (hunk 2 above)
Comment (siwane) [reply src/qtism/data/storage/xml/marshalling/OutcomeDeclarationMarshaller.php]: +1

**Comment by siwane** [comment]
Could be nice to keep the logic of `null` value with a `hasExternalScored` method, it is not required just a proposal

File: src/qtism/data/state/OutcomeDeclaration.php [hunk 4]
```diff
@@ -361,4 +370,26 @@ public function getComponents()
 
         return new QtiComponentCollection($comp);
     }
+
+    /**
+     * Set external score attribute to determine how scoring should be proceed
+     *
+     * @param string $externalScored
+     */
+    public function setExternalScored($externalScored = null)
```
Comment (siwane) [inline src/qtism/data/state/OutcomeDeclaration.php]: Can u be fluent please (return $this)

**Pull request merged**